src/k14s_releases_service.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  GitHubReleasesService,
+  Octokit,
+  AppInfo,
+  ReposListReleasesParameters,
+  DownloadInfo,
+  GitHubDownloadMeta
+} from '@jbrunton/gha-installer'
+import {
+  ActionsCore,
+  Environment,
+  FileSystem
+} from '@jbrunton/gha-installer/lib/interfaces'
+import * as crypto from 'crypto'
+import * as path from 'path'
+import * as fs from 'fs'
+import * as core from '@actions/core'
+
+export class K14sReleasesService extends GitHubReleasesService {
+  private _fs: FileSystem
+
+  constructor(
+    core: ActionsCore,
+    env: Environment,
+    fs: FileSystem,
+    octokit: Octokit
+  ) {
+    super(core, env, octokit, getRepo, getAssetName)
+    this._fs = fs
+  }
+
+  onFileDownloaded(
+    path: string,
+    info: DownloadInfo<GitHubDownloadMeta>,
+    core: ActionsCore
+  ): void {
+    this.verifyChecksum(path, info, core)
+  }
+
+  private verifyChecksum(
+    downloadPath: string,
+    info: DownloadInfo<GitHubDownloadMeta>,
+    core: ActionsCore
+  ) {
+    const digest = this.computeDigest(downloadPath)
+    const assetName = path.basename(info.url)
+    const expectedChecksum = `${digest}  ./${assetName}`
+    const releaseNotes = info.meta.release.body
+    if (releaseNotes.includes(expectedChecksum)) {
```

Comment:
Where we verify the checksum. We compute the digest and then check that a string that matches the output of `shasum -a 256 ./<filename>` exists in the readme notes for the release.